sim/src/main/cc/firesim/firesim_top.cc

```diff
@@ -45,7 +50,7 @@ firesim_top_t::firesim_top_t(simif_t &simif,
 
   // Add functions you'd like to periodically invoke on a paused simulator here.
   if (profile_interval) {
-    register_task(0, [&] {
+    register_task(0, [&, profile_interval] { // capture profile_interval by copy otherwise you will get a invalid optional
```

Comment:
```suggestion
    // capture profile_interval by value, since its lifetime is bound to firesim_top_t's constructor
    // and its value is needed for the duration of the simulation.
    register_task(0, [&, profile_interval] {
```
You'll need to run clang-format on this.

Reply:
@nandor Do modern sanitizers check for this sort of binding issue?

Reply:
Apparently not. This is unfortunate.

Reply:
LLVM/CIRCT allow implicit captures, but we could go for a style guide decision by disabling them and enforcing explicit captures.